### PR TITLE
WIP: [xacro] use `PYTHON_EXECUTABLE` and fix `OUTPUT_NAME`

### PR DESCRIPTION
**Pull request by nim65s**
Hi,

I had to do this to get xacro working with python3, but I would understand that we don't need to use the same python interpreter to use xacro and for the rest of the project, so this is open for discussions.

### CODE REVIEW

File: xacro.cmake [hunk 1]
```diff
@@ -87,5 +87,5 @@ MACRO(GENERATE_URDF_FILE FILENAME EXTENSION)
     SET(_XACRO_FILE_ ${CMAKE_CURRENT_SOURCE_DIR}/${FILENAME}.xacro)
   ENDIF (NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${FILENAME}.xacro)
 
-  RUN_XACRO(${_XACRO_FILE_} ${CMAKE_CURRENT_BINARY_DIR}/${OUTPUT_FILE})
+  RUN_XACRO(${_XACRO_FILE_} ${CMAKE_CURRENT_BINARY_DIR}/${FILENAME}.${EXTENSION})
```
Comment (jmirabel) [inline xacro.cmake]: How the hell did this ever worked !

File: xacro.cmake [hunk 2]
```diff
@@ -40,8 +40,8 @@ MACRO(RUN_XACRO INPUT OUTPUT)
 
   ADD_CUSTOM_COMMAND(
     OUTPUT ${OUTPUT}
-    COMMAND ${_xacro_py}
-    ARGS ${XACRO_OPTIONS} -o ${OUTPUT} ${INPUT}
+    COMMAND ${PYTHON_EXECUTABLE}
```
Comment (jmirabel) [inline xacro.cmake]: Could you add a FINDPYTHON at the beginning of the macro ?

**Comment by jmirabel** [comment]
I am not sure whether this is a good idea.
What issue did you get ?

File: xacro.cmake (hunk 1 above)
Comment (nim65s) [reply xacro.cmake]: This is an excellent question. I guess a commit forgot to update this too at some point.

Comment (nim65s) [reply xacro.cmake]: or maybe xacro is able to understand that when the output is a directory, it just has to put a file with `${FILENAMEE}.urdf` in this directory.

**Comment by nim65s** [comment]
I installed xacro from source from robotpkg while robotpkg was configured in python3, and xacro's main script started with `#!/usr/bin/env python`, and contained an `import xacro`, but the `xacro` python module was nowhere in the `sys.path` of the main `python` of my system, which was python2.

So, more generally, xacro can be installed with a python version which is independent from the OS, and raise this issue… But it can also (and this might occur more often) be installed with a python version different from `PYTHON_EXECUTABLE`, so maybe we would need a `XACRO_PYTHON_EXECUTABLE`, but this looks overkill.

I don't have any good solution right now, this is why I just wanted to get opinions.

File: xacro.cmake (hunk 2 above)
Comment (nim65s) [reply xacro.cmake]: Sure.
Will do if we decide do keep it :)

**Comment by nim65s** [comment]
rebased

**Comment by nim65s** [comment]
#348 was included by mistake during the rebase ; please don't merge :)

**Comment by nim65s** [comment]
Actually, this works well: https://github.com/ros/xacro/blob/1.13.5/cmake/xacro-extras.cmake

eg.
```cmake
FOREACH(F ${XACRO_FILES})
  xacro_add_xacro_file(${F}.urdf.xacro)
  INSTALL(FILES ${XACRO_OUTPUT_FILE} DESTINATION …)
ENDFOREACH(F ${XACRO_FILES}) 
```

So I suggest to deprecate this xacro.cmake file :)

**Comment by jmirabel** [comment]
> ```cmake
> ```cmake
> 
> ```
> 
> 
> So I suggest to deprecate this xacro.cmake file :)
> ```

Agreed.

**Comment by nim65s** [comment]
replaced by #490

**Pull request closed without merging**